[Lim Danqi] iP

limdanqi opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

#

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        new Duke("data/duke.txt").run();
    }
}
```

Commit:
Add Gradle support

limdanqi committed:
Level-1

limdanqi committed:
Level-2

limdanqi committed:
Level-3

limdanqi committed:
Level-4

limdanqi committed:
A-TextUiTesting

limdanqi committed:
Level-5

limdanqi committed:
Level-6

limdanqi committed:
Level-7

limdanqi committed:
Implemented Level-8

limdanqi committed:
Merge branch 'branch-Level-8'

limdanqi committed:
Merged branch-Level-7 and branch-Level-8

limdanqi committed:
Implemented A-MoreOOP

limdanqi committed:
Implemented A-Packages

limdanqi committed:
Implemented A-JUnit

limdanqi committed:
Implemented branch-Level-9

limdanqi committed:
Implemented branch-A-JavaDoc

limdanqi committed:
Implemented branch-A-CodingStandard

limdanqi committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java

limdanqi committed:
Merge remote-tracking branch 'origin/branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/Task.java

limdanqi committed:
Merged branches Level-9, A-JavaDoc, A-CodingStandard

limdanqi committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

limdanqi committed:
Implement A-Gradle

yongchuann commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,75 @@
+        }
+
+        sc.close();
+
```
Random spaces here

limdanqi replied:
Ok thank you! I would make the appropriate changes :D

yongchuann commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,76 @@
+        return commandTypes;
+    }
+
+    //todo command already present
```
Did you forget to javadoc for this?

yongchuann commented on src/main/java/duke/data/DateAndTime.java:
```diff
@@ -0,0 +1,124 @@
+                    + dateAndTimeFromCommand.substring(lastSlashIndex - 2, lastSlashIndex) + "-"
+                    + dateAndTimeFromCommand.substring(lastSlashIndex - 5, lastSlashIndex - 3);
+        }
+
```
Extra Space

yongchuann commented on src/main/java/duke/data/exceptions/InvalidDateAndTimeException.java:
```diff
@@ -0,0 +1,37 @@
+        }
+    }
+    //todo message for date or time being optional?
+
```
Extra Space

yongchuann commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,216 @@
+
+    }
+
+    private String getTaskName(String command, String commandType) throws InvalidInputException, EmptyTaskDescriptionException {
```
This is more than 120 characters so i think you maybe need to indent it!

yongchuann commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,216 @@
+        }
+    }
+
+    private String getDateAndTime(String message, String taskType) throws InvalidDateAndTimeException, InvalidInputException {
```
Also above 120 characters!

yongchuann commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,216 @@
+                break; //task number string complete
+            } else if (Character.isDigit(currentChar)) {
+                numberString += message.charAt(i);
+            } else {}
```
Will you be adding something here?

yongchuann reviewed (commented):
Looks good to me!
Just some small things like inconsistent spacing especially at the return block

limdanqi committed:
Implement A-CheckStyle

limdanqi commented:
> Looks good to me!
> Just some small things like inconsistent spacing especially at the return block

limdanqi committed:
Implement Level-10

limdanqi committed:
Added parts of GUI incorporation

luodan01 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,216 @@
+ * Represents a parser that makes sense of what the user input is.
+ */
+public class Parser {
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String EVENT = "event";
+    private static final String EXIT = "bye";
+    private static final String LIST = "list";
+    private static final String DONE = "done";
+    private static final String DELETE = "delete";
+    private static final String FIND = "find";
```
Perhaps it would be better to implement these as enums?

limdanqi replied:
Ok thank you for reviewing :)!

luodan01 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,161 @@
+package duke.storage;
+
+import java.io.*;
```
Perhaps it would be better to import the relevant classes explicitly?

luodan01 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,161 @@
+    private Task getTaskFromString(String taskString) {
+        String taskType = getTaskTypeFromString(taskString);
+        String taskName = getTaskNameFromString(taskString);
+        boolean taskStatus = getTaskStatusFromString(taskString);
```
Perhaps it would be better to name this ```isDone``` (same as in the ```Task``` class) since it is a boolean?

luodan01 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,34 @@
+        return taskList.get(taskPosition);
+    }
+
+    public int size() {
```
Perhaps ```size``` could be named as ```getSize``` instead?

luodan01 reviewed (commented):
LGTM! I really like how you organise your code and give meaningful variable names.

limdanqi committed:
Added Part 2 of GUI incorporation

limdanqi committed:
Implement A-Assertions

Edited the code to add in assertions.

limdanqi committed:
Implement C-Update

There is no edit command currently.

Edit commands are added so that users can edit task names or dates for
instance.

limdanqi committed:
Implement A-CodeQuality

Improved the code quality where necessary.

limdanqi committed:
Merge pull request #1 from limdanqi/branch-A-Assertions

Implement A-Assertions

limdanqi committed:
Merge branch 'master' into branch-A-CodeQuality

limdanqi committed:
Merge pull request #2 from limdanqi/branch-A-CodeQuality

Implement A-CodeQuality

limdanqi committed:
Set theme jekyll-theme-minimal

limdanqi committed:
Merge branch 'branch-C-Update'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/ui/Tag.java

limdanqi committed:
Implement A-UserGuide and merge C-Update

Implemented the user guide in the README file and merged branch-C-update
 to the master branch.

limdanqi committed:
Update README.md

limdanqi committed:
Update README.md

limdanqi committed:
Update README.md

limdanqi committed:
Update README.md

limdanqi committed:
Update README.md

limdanqi committed:
Implement A-BetterGui

Implemented A-BetterGui and cleaned up the code.

limdanqi committed:
Merge remote-tracking branch 'origin/master'

limdanqi committed:
Update README.md

Updated README.md to include the product screenshot.

limdanqi committed:
Update README.md

Updated README.md to include the product screenshot.

limdanqi committed:
Update README.md
